### PR TITLE
fix for reading jpeg-xr images

### PR DESCRIPTION
**Pull request by dimin**

### CODE REVIEW

File: src/tiffimage_int.cpp [hunk 1]
```diff
@@ -2032,7 +2032,9 @@ namespace Exiv2 {
         else {
             return false;
         }
-        if (tag_ != getUShort(pData + 2, byteOrder_)) return false;
+        //if (tag_ != getUShort(pData + 2, byteOrder_)) return false;
```
Comment (kmilos) [inline src/tiffimage_int.cpp:2035]:
```suggestion
```

File: src/tiffimage_int.cpp [hunk 2]
```diff
@@ -2032,7 +2032,9 @@ namespace Exiv2 {
...
+        uint16_t t = getUShort(pData + 2, byteOrder_);
+        if (t != 444 && tag_ != t) return false; // dima: 444 is for the JPEG-XR
```
Comment (kmilos) [inline src/tiffimage_int.cpp:2037]:
```suggestion
        if (t != 444 && tag_ != t) return false; // 444 is for the JPEG-XR
```

**Comment by kmilos** [comment]
Thanks. Would you also be able to add a test case for this please?

**Comment by kevinbackhouse** [comment]
@dimin: Thanks for the bug fix! Please could you add a test image for this? Adding a test is quite easy: you just need to add an image file to `test/data` and a python script to `tests/bugfixes/github`. [Here's a link](https://github.com/Exiv2/exiv2/commit/0f0885b0d6251f1dae3cc6d013d600fae26b3cfd#diff-9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08) to a recent example.

**Comment by postscript-dev** [comment]
Please can you make a very minor change to the manpage, in the [FILE TYPE](https://github.com/Exiv2/exiv2/blob/main/man/man1/exiv2.md#3-file-types) section. Simply add an extra row to the table for JPEG-XL, and explain what features are supported. Thanks.

**Comment by dimin** [comment]
Added the JPEG-XR format to the manpage. I can also contribute a test file but would prefer if you guys could add a test case for it?

**Comment by kmilos** [comment]
Sorry, thinking about this some more, I think we should probably take this out to a proper new format and source file, like the other TIFF-derived formats (see ORF, RW2), rather than just simply piggy-backing TIFF base classes.

**Comment by kevinbackhouse** [comment]
I see that @dimin added a test file in a separate PR: #1946

**Comment by piponazo** [comment]
Hi guys. Should we get this merged as it is so that we have at least the minimum functionality needed by the contributor?

**Comment by kmilos** [comment]
I'd prefer to handle it as other TIFF derivatives.

**Comment by piponazo** [comment]
> I'd prefer to handle it as other TIFF derivatives.

I totally agree that this should be the end goal, as you said there might be more to JPEG-XR the container than just the header difference. On the other hand it is difficult to ask for a new contributor to undertake such task since it will take a considerable amount of time. And this change is actually enabling the parsing of JPEG-XR files for him (and probably other potentially interested people).

What about doing the following?
- Accept & merge this change 
- Creating a new test file in a separated PR with the test file provided at #1946. I can take care of that.
- Open a new issue to implement the support for JPEG-XR in a separate class and investigate what else we would need to have a proper support for that format. (This issue could be taken by anybody interested in the topic).

IMHO that would be a pragmatic & realistic approach.

**Comment by kmilos** [comment]
Ok, let's go with that approach, we just need to make sure we don't forget what needs to be removed later.

**Review by piponazo — APPROVED** [review]

**Comment by piponazo** [comment]
I created 2 issues. I'll work on the test case when I have some time 😉

**Pull request merged**